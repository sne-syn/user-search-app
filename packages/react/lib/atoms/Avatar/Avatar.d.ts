import { FC } from "react";
interface AvatarProps {
    src: string;
}
declare const Avatar: FC<AvatarProps>;
export default Avatar;
