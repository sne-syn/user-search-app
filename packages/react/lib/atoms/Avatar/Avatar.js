import React from 'react';

const Avatar = ({ src }) => {
    return React.createElement("img", { className: "dse-avatar", src: src });
};

export { Avatar as default };
//# sourceMappingURL=Avatar.js.map
