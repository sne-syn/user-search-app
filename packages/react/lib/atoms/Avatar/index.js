export { default } from './Avatar.js';
//# sourceMappingURL=index.js.map
