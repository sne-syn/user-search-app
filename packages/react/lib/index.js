export { default as Avatar } from './atoms/Avatar/Avatar.js';
export { default as Button } from './atoms/Button/Button.js';
export { default as CardSubtitle } from './atoms/CardSubtitle/CardSubtitle.js';
export { default as CardTitle } from './atoms/CardTitle/CardTitle.js';
export { default as Icon } from './atoms/Icon/Icon.js';
export { default as Input } from './atoms/Input/Input.js';
export { default as Logo } from './atoms/Logo/Logo.js';
export { default as DateInfo } from './atoms/DateInfo/DateInfo.js';
export { default as DefinitionList } from './atoms/DefinitionList/DefinitionList.js';
export { default as ErrorMessage } from './atoms/ErrorMessage/ErrorMessage.js';
export { default as TextBlock } from './atoms/TextBlock/TextBlock.js';
export { default as ThemeToggler } from './atoms/ThemeToggler/ThemeToggler.js';
export { default as VisuallyHidden } from './atoms/VisuallyHidden/VisuallyHidden.js';
export { default as Wrapper } from './atoms/Wrapper/Wrapper.js';
//# sourceMappingURL=index.js.map
