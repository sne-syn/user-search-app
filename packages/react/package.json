{
  "name": "@user-search-app/react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "build": "rollup -c",
    "dev": "yarn build --watch",
    "test": "jest --verbose",
    "test:watch": "yarn test --watch"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.18.10",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@rollup/plugin-image": "^2.1.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.3.0",
    "@types/jest": "^28.1.6",
    "@types/react": "^18.0.15",
    "jest": "^28.1.3",
    "jest-environment-jsdom": "^28.1.3",
    "react": "^18.2.0",
    "rollup": "^2.77.0",
    "rollup-plugin-img": "^1.1.0",
    "rollup-plugin-typescript2": "^0.32.1",
    "typescript": "^4.7.4"
  }
}
