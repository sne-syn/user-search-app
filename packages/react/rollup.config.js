import Ts from "rollup-plugin-typescript2";
import image from "@rollup/plugin-image";

export default {
  input: [
    "src/index.ts",
    "src/atoms/Avatar/index.ts",
    "src/atoms/Button/index.ts",
    "src/atoms/CardSubtitle/index.ts",
    "src/atoms/CardTitle/index.ts",
    "src/atoms/Icon/index.ts",
    "src/atoms/Input/index.ts",
    "src/atoms/Link/index.ts",
    "src/atoms/DefinitionList/index.ts",
    "src/atoms/DateInfo/index.ts",
    "src/atoms/Logo/index.ts",
    "src/atoms/TextBlock/index.ts",
    "src/atoms/ThemeToggler/index.ts",
    "src/atoms/VisuallyHidden/index.ts",
    "src/atoms/Wrapper/index.ts",
  ],
  output: {
    dir: "lib",
    format: "esm",
    sourcemap: true,
  },
  plugins: [
    Ts(),
    image({
      extensions: /\.(png|jpg|jpeg|gif|svg)$/,
      limit: 10000,
    }),
  ],
  preserveModules: true,
  external: ["react", "@ds.e/foundation"],
};
