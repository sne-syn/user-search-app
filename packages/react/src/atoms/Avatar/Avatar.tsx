import React, { FC } from "react";

interface AvatarProps {
  src: string;
  alt: string;
  role?: "presentation" | "img";
  style?: React.CSSProperties;
}
const Avatar: FC<AvatarProps> = ({ src, alt, style, role = "img" }) => {
  return (
    <img
      style={style}
      className="dse-avatar"
      src={src}
      alt={alt}
      role={role}
      data-testid="avatar"
    />
  );
};

export default Avatar;
