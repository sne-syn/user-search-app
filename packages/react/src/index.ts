import Avatar from "./atoms/Avatar";
import Button from "./atoms/Button";
import CardSubtitle from "./atoms/CardSubtitle";
import CardTitle from "./atoms/CardTitle";
import Input from "./atoms/Input";
import ListContent from "./atoms/ListContent";
import ListIcon from "./atoms/ListIcon";
import Logo from "./atoms/Logo";
import DateInfo from "./atoms/DateInfo";
import DefinitionList from "./atoms/DefinitionList";
import ErrorMessage from "./atoms/ErrorMessage";
import TextBlock from "./atoms/TextBlock";
import ThemeToggler from "./atoms/ThemeToggler";
import VisuallyHidden from "./atoms/VisuallyHidden";
import Wrapper from "./atoms/Wrapper";

export {
  Avatar,
  Button,
  CardSubtitle,
  CardTitle,
  Input,
  ListContent,
  ListIcon,
  Logo,
  DateInfo,
  DefinitionList,
  ErrorMessage,
  TextBlock,
  ThemeToggler,
  VisuallyHidden,
  Wrapper,
};
