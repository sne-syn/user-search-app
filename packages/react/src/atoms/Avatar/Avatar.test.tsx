import React from "react";
import Avatar from "./Avatar";
import { render, screen } from "@testing-library/react";
const SRC = "https://avatars.githubusercontent.com/u/583231?v=3";
test("renders avatar", () => {
  render(<Avatar src={SRC} />);
  expect(screen.findByTestId("avatar")).toBeInTheDocument;
  expect(screen.findByTestId("avatar")).toMatchSnapshot;
});
