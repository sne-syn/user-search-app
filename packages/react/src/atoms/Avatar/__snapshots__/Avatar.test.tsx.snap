// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders avatar 1`] = `
<img
  alt="User's avatar"
  class="dse-avatar"
  data-testid="avatar"
  src="https://avatars.githubusercontent.com/u/583231?v=3"
/>
`;

exports[`Avatar renders avatar with style provided 1`] = `
<img
  alt="User's avatar"
  class="dse-avatar"
  data-testid="avatar"
  src="https://avatars.githubusercontent.com/u/583231?v=3"
  style="width: 100px;"
/>
`;
